File: generated_code/optimized/memory/java/5.java
import java.util.*;

class Solution {
    /**
    Insert a number 'delimeter' between every two consecutive elements of input list `numbers'
    >>> intersperse(List.of(), 4)
    []
    >>> intersperse(Arrays.asList(1, 2, 3), 4)
    [1, 4, 2, 4, 3]
     */
    public List<Integer> intersperse(List<Integer> numbers, int delimiter) {
        int n = numbers.size();
        if (n == 0) {
            return Collections.emptyList();
        }
        ArrayList<Integer> result = new ArrayList<>(n * 2 - 1);
        Iterator<Integer> it = numbers.iterator();
        result.add(it.next());
        while (it.hasNext()) {
            result.add(delimiter);
            result.add(it.next());
        }
        return result;
    }
}

File: generated_code/optimized/balanced/minimal/cpp/5.cpp
#include <vector>
using namespace std;

vector<int> intersperse(vector<int> numbers, int delimeter) {
    const size_t n = numbers.size();
    if (n <= 1) return numbers;

    vector<int> out(2 * n - 1);
    out[0] = numbers[0];

    for (size_t i = 1; i < n; ++i) {
        out[2 * i - 1] = delimeter;
        out[2 * i] = numbers[i];
    }

    return out;
}
